sumpy/p2e.py

```diff
@@ -80,24 +81,40 @@ def get_loopy_instructions(self):
         from sumpy.assignment_collection import SymbolicAssignmentCollection
         sac = SymbolicAssignmentCollection()
 
-        coeff_names = [
-            sac.assign_unique("coeff%d" % i, coeff_i)
+        coeff_names = []
+        code_transformers = [self.expansion.get_code_transformer()]
+        for knl_idx, kernel in enumerate(self.kernels):
+            code_transformers.append(kernel.get_code_transformer())
             for i, coeff_i in enumerate(
-                self.expansion.coefficients_from_source(avec, None, rscale, sac))]
+                self.expansion.coefficients_from_source(avec, None, rscale,
+                     sac, kernel=kernel)
+            ):
+                sac.add_assignment(f"coeff{i}_{knl_idx}", coeff_i)
+                coeff_names.append(f"coeff{i}_{knl_idx}")
 
         sac.run_global_cse()
 
         from sumpy.codegen import to_loopy_insns
         return to_loopy_insns(
                 sac.assignments.items(),
                 vector_names={"a"},
-                pymbolic_expr_maps=[self.expansion.get_code_transformer()],
+                pymbolic_expr_maps=code_transformers,
```

Comment:
This will implicitly apply all transformers to the code out of all kernels. How can we be sure that that's OK?

Reply:
> This will implicitly apply all transformers to the code out of all kernels. How can we be sure that that's OK?

Yes, that's a valid concern, but I'm not sure how to fix that.

Reply:
Same issue is in P2P in master as well. https://github.com/inducer/sumpy/blob/db59a26e20cd3f210aa64ee91bd15fb4f35016e2/sumpy/p2p.py#L100-L101

Reply:
Fair enough, let's live with that for now.